### PR TITLE
solution

### PR DESCRIPTION
**Pull request by alim2709**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,19 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
```
Comment (ArsenPidhoretskyi) [inline app/main.py:7]:
```suggestion
        self.people[name] = self
```
It is not needed. You can just use self.people

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for person in people:
+        result += {Person(person.get("name"), person.get("age"))}
```
Comment (ArsenPidhoretskyi) [inline app/main.py:13]: Use list comprehension

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+    for i, person in enumerate(people):
```
Comment (ArsenPidhoretskyi) [inline app/main.py:14]: Why you need i

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,19 @@
...
+        if "wife" in person and person.get("wife") is not None:
+            result[i].wife = Person.people.get(person["wife"])
```
Comment (ArsenPidhoretskyi) [inline app/main.py:16]: Objects are mutable. Then if you change it inside
Person.people it will affect result also

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,19 @@
...
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for person in people:
+        result += {Person(person.get("name"), person.get("age"))}
+    for i, person in enumerate(people):
+        if "wife" in person and person.get("wife") is not None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:15]:
```suggestion
        if person.get("wife") is not None:
```
The same with husband

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 6]
```diff
@@ -1,8 +1,22 @@
...
+    result = [
+        Person(person.get("name"), person.get("age")) for person in people
+    ]
+    i = 0
+    for person in people:
+        if person.get("wife") is not None:
+            result[i].wife = Person.people.get(person["wife"])
+        if person.get("husband") is not None:
+            result[i].husband = Person.people.get(person["husband"])
+        i += 1
```
Comment (Bohdan-Salamakha) [inline app/main.py:20]: use for i in range(len(people)):
to avoid i = 0, i+= 1

u don't need to use C-like counters in Python

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**